src/topology/instances/matrix.lean

```diff
@@ -310,17 +322,72 @@ lemma summable.matrix_block_diagonal [decidable_eq p] {f : X → p → matrix m
   summable (λ x, block_diagonal (f x)) :=
 hf.has_sum.matrix_block_diagonal.summable
 
+lemma summable_matrix_block_diagonal [decidable_eq p] {f : X → p → matrix m n R} :
+  summable (λ x, block_diagonal (f x)) ↔ summable f :=
+(summable.map_iff_of_left_inverse
+  (matrix.block_diagonal_add_monoid_hom m n p R) (matrix.block_diag_add_monoid_hom m n p R)
+  (by exact continuous.matrix_block_diagonal continuous_id)
+  (by exact continuous.matrix_block_diag continuous_id)
+  (λ A, block_diag_block_diagonal A) : _)
+
+lemma matrix.block_diagonal_tsum [decidable_eq p] [t2_space R] {f : X → p → matrix m n R} :
+  block_diagonal (∑' x, f x) = ∑' x, block_diagonal (f x) :=
+begin
+  by_cases hf : summable f,
+  { exact hf.has_sum.matrix_block_diagonal.tsum_eq.symm },
+  { have hft := summable_matrix_block_diagonal.not.mpr hf,
+    rw [tsum_eq_zero_of_not_summable hf, tsum_eq_zero_of_not_summable hft],
+    exact block_diagonal_zero },
+end
+
+lemma has_sum.matrix_block_diag {f : X → matrix (m × p) (n × p) R}
+  {a : matrix (m × p) (n × p) R} (hf : has_sum f a) :
+  has_sum (λ x, block_diag (f x)) (block_diag a) :=
+(hf.map (block_diag_add_monoid_hom m n p R) $ continuous.matrix_block_diag continuous_id : _)
+
+lemma summable.matrix_block_diag {f : X → matrix n n R} (hf : summable f) :
+  summable (λ x, diag (f x)) :=
+hf.has_sum.matrix_diag.summable
```

Comment:
Is this the lemma you intended? Or should it be
```suggestion
lemma summable.matrix_block_diag {f : X → matrix n n R} (hf : summable f) :
  summable (λ x, block_diag (f x)) :=
hf.has_sum.matrix_block_diag.summable
```

Reply:
Nice catch! The type of `f` was wrong too; now fixed.